pallets/delegating/src/lib.rs

```diff
@@ -0,0 +1,494 @@
+        NoneValue,
+        /// Errors should have helpful documentation associated with them.
+        StorageOverflow,
+        AlreadyDelegated,
+        NotDelegate,
+        CreditLocked,
+        NoCreditLedgerData,
+        NotRightEra,
+        CreditScoreTooLow,
+        NonCandidateValidator,
+        NotInCandidateValidator,
+        SelectTooManyValidators,
+        SelectNoValidator,
+        DelegateMoreCreditScoreThanOwn,
+        InvalidEraToReward,
+    }
+}
+
+// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+// These functions materialize as "extrinsics", which are often compared to transactions.
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+
+        // Events must be initialized if they are used by the pallet.
+        fn deposit_event() = default;
+
+        /// delegate credit score to validators in vec equally
+        #[weight = 10_000]
+        pub fn delegate(origin, validators: Vec<T::AccountId>) -> dispatch::DispatchResult {
+            // check signature
+            let controller = ensure_signed(origin)?;
+
+            // check credit pass threshold
+            if T::CreditInterface::pass_threshold(controller.clone(), 0) == false{
+                error!("Credit score is to low to delegating a validator!");
+                Err(Error::<T>::CreditScoreTooLow)?
+            }
+
+            // check validators size
+            if validators.len() > MAX_VALIDATORS_CAN_SELECTED {
+                Err(Error::<T>::SelectTooManyValidators)?
+            }
+            if validators.len() == 0{
+                Err(Error::<T>::SelectNoValidator)?
+            }
+
+            // check if controller has call delegated
+            if <DelegatedToValidators<T>>::contains_key(controller.clone()){
+                Err(Error::<T>::AlreadyDelegated)?
+            }
+
+            // check target validators in candidate_validators
+            let candidate_validators = <CandidateValidators<T>>::get().unwrap();
+            for validator in validators.clone() {
+                if !candidate_validators.contains(&validator){
+                    error!("Validator AccountId  isn't in candidateValidators");
+                    Err(Error::<T>::NotInCandidateValidator)?
+                }
+            }
+
+            // get avg score to validators
+            let validators_vec = Self::cut_credit_score(controller.clone(), validators.clone());
+            Self::_delegate(controller.clone(), validators_vec);
+
+            let credit_delegate_info = CreditDelegateInfo{
+                delegator: controller.clone(),
+                score: T::CreditInterface::get_credit_score(controller.clone()).unwrap(),
```

Comment:
is it possible for the option to return None so that crash by unwrap?

Reply:
`            // check credit pass threshold
            if T::CreditInterface::pass_threshold(controller.clone(), 0) == false{
                error!("Credit score is to low to delegating a validator!");
                Err(Error::<T>::CreditScoreTooLow)?
            }
`
It's impossible for get_credit_score to return None  here. We have called pass_threshold() in line 150